internal/providers/detect.go

```diff
@@ -70,67 +71,74 @@ func Detect(ctx *config.ProjectContext, includePastResources bool) (schema.Provi
 		}
 
 		return h, nil
+	case "terragrunt_dir":
+		h := terraform.NewTerragruntHCLProvider(ctx, includePastResources)
+		if err := validateProjectForHCL(ctx, path); err != nil {
+			return h, err
+		}
+
+		return h, nil
+	case "terraform_plan_json":
+		return terraform.NewPlanJSONProvider(ctx, includePastResources), nil
+	case "terraform_plan_binary":
+		return terraform.NewPlanProvider(ctx, includePastResources), nil
+	case "terraform_cli":
+		return terraform.NewDirProvider(ctx, includePastResources), nil
+	case "terragrunt_cli":
+		return terraform.NewTerragruntProvider(ctx, includePastResources), nil
+	case "terraform_state_json":
+		return terraform.NewStateJSONProvider(ctx, includePastResources), nil
+	case "cloudformation":
+		return cloudformation.NewTemplateProvider(ctx, includePastResources), nil
 	}
 
+	return nil, fmt.Errorf("Could not detect path type for '%s'", path)
+}
+
+func autoDetectProjectType(path string) string {
 	if isCloudFormationTemplate(path) {
-		return cloudformation.NewTemplateProvider(ctx, includePastResources), nil
+		return "cloudformation"
 	}
 
 	if isTerraformPlanJSON(path) {
-		return terraform.NewPlanJSONProvider(ctx, includePastResources), nil
+		return "terraform_plan_json"
 	}
 
 	if isTerraformStateJSON(path) {
-		return terraform.NewStateJSONProvider(ctx, includePastResources), nil
+		return "terraform_state_json"
 	}
 
 	if isTerraformPlan(path) {
-		return terraform.NewPlanProvider(ctx, includePastResources), nil
+		return "terraform_plan_binary"
 	}
 
-	if isTerragruntDir(path) {
-		return terraform.NewTerragruntProvider(ctx, includePastResources), nil
+	if isTerragruntNestedDir(path, 5) {
+		return "terragrunt_dir"
 	}
 
 	if isTerraformDir(path) {
-		return terraform.NewDirProvider(ctx, includePastResources), nil
-	}
-
-	if isTerragruntNestedDir(path, 5) {
-		return terraform.NewTerragruntProvider(ctx, includePastResources), nil
+		return "terraform_dir"
 	}
 
-	return nil, fmt.Errorf("Could not detect path type for '%s'", path)
+	return ""
 }
 
 func validateProjectForHCL(ctx *config.ProjectContext, path string) error {
-	if isTerragruntDir(path) || isTerragruntNestedDir(path, 5) {
-		return &ValidationError{
-			err: "Terragrunt projects are not yet supported by the parse HCL option.",
-		}
-	}
-
-	if isCloudFormationTemplate(path) {
-		return &ValidationError{
-			err: "CloudFormation projects are not yet supported by the parse HCL option.",
-		}
-	}
-
-	if isTerraformPlanJSON(path) {
+	if ctx.ProjectConfig.TerraformPlanFlags != "" {
 		return &ValidationError{
-			err: "Path type cannot be a Terraform plan JSON file when using the parse HCL option\n\nSet --path to a Terraform directory.",
+			warn: "terraform-plan-flags is deprecated with Terraform HCL parsing, use --terraform-var or --terraform-var-file instead.",
```

Comment:
maybe this should say "not supported" over "deprecated" as HCL doesn't use `terraform-plan-flags`

Reply:
I'm not sure. Technically it is supported at the moment, since we'll parse out the `-var` and `-var-file` flags from it.

Reply:
Ah, in that case deprecated works